File: client/src/model/ConstructionElement.ts
import ProfileClass from './ProfileClass'
import Steel from './Steel'

import Profile from './Profile'

interface IConstructionElement {
	id: number
	profileClass: ProfileClass
	profile: Profile
	steel: Steel
	length: number
	arithmeticExpression: string
}

class ConstructionElement {
	id: number
	profileClass: ProfileClass
	profile: Profile
	steel: Steel
	length: number
	arithmeticExpression: string

	constructor(obj?: IConstructionElement) {
		this.id = (obj && obj.id) || 0
		this.profileClass = (obj && obj.profileClass) || null
		this.profile = (obj && obj.profile) || null
		this.steel = (obj && obj.steel) || null
		this.length = (obj && obj.length) || 0
		this.arithmeticExpression = (obj && obj.arithmeticExpression) || ''
	}
}

export default ConstructionElement

File: server/DocumentsKM/Models/ConstructionElement.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DocumentsKM.Models
{
    // Элемент вида конструкции
    public class ConstructionElement
    {
        [Key]
        public Int32 Id { get; set; }

        // Вид конструкции
        [Required]
        [ForeignKey("ConstructionId")]
        public virtual Construction Construction { get; set; }

        // Профиль
        [Required]
        [ForeignKey("ProfileId")]
        public virtual Profile Profile { get; set; }

        // Марка стали
        [Required]
        [ForeignKey("SteelId")]
        public virtual Steel Steel { get; set; }

        // Длина
        [Required]
        [Range(0, 65535, ErrorMessage = "Value should be greater than or equal to 0")]
        public float Length { get; set; }

        // Арифметическое выражение
        [MaxLength(255)]
        public string ArithmeticExpression { get; set; }
    }
}
